src/mirage_kv.mli

```diff
@@ -199,6 +210,13 @@ module type RW = sig
      {!batch} operation, where durability will be guaranteed at the
      end of the batch. *)
 
+  val write: t -> key -> int -> string -> (unit, write_error) result Lwt.t
+  (** [write t k o v] replaces the part of the value bound to [k] in
+     [t] at offset [o] with [v].
+
+      The result is [Error (`Value_expected k)] if [k] refers to a
+     dictionary in [t]. *)
```

Comment:
so if the value of `foo` is 4 bytes long, and a `set_partial` at offset 2 is used to write 1 byte, will the new value of `foo` be still 4 bytes, or 3 bytes?

I.e. foo -> "OOOO", set_partial foo 2 "b", will now foo be "OObO" or "OOb"?